### DISCUSSION TITLE
Why does ctrl+shift+z stop all jobs?

### DISCUSSION THREAD

**Discussion by omentic** [kovidgoyal/kitty#4534] [Q&A] [answered]
I unfortunately tend to press this occasionally out of habit (redo on some systems) and it crashes (Neo)Vim and (Neo)Vim exclusively. I assume this is a terminal-level keybinding because it didn't happen when I tested it in Alacritty.

What's the background behind this? I was unable to find anything resembling ctrl+shift+z in my config file, so it seems to be part of some deeper part of kitty.

**Comment by page-down** [comment]
Are you using kitty version 0.24+? This is a default shortcut related to shell integration.

By `crash` and `stop all jobs` you describe, do you mean the neovim process exiting? Or is it that the kitty process crashes and the window simply disappears?

How to reproduce this issue? I press this shortcut under macOS, it doesn't crash.

**Comment by kovidgoyal** [comment, marked as answer]
ctrl+shift+z jumps to the previous prompt, but it will be a no-op when
nvim is running since nvim switches to the alternate screen, where there
is no scrollback.

So what is crashing and what steps reproduce this crash.

**Reply by omentic** [reply]
Here is an asciinema recording: https://asciinema.org/a/qmMrH0BlWfQgKUNPoSfPu51AV
I'm not yet certain of the steps to reproduce.

**Reply by omentic** [reply]
And what do you mean by the previous prompt? I haven't noticed that ctrl+shift+z does anything outside of Neovim. Maybe my shell integration is disabled?

**Reply by kovidgoyal** [reply]
See https://sw.kovidgoyal.net/kitty/shell-integration

**Reply by kovidgoyal** [reply]
Try running kitty with --debug-input that will tell you what is happening when you press ctrl+shift+z is it doing an action in kitty or being sent to the child

**Reply by kovidgoyal** [reply]
Actually I am guessing you are running a version of kitty before 0.24 in which case kitty will either send ^Z or ^[[122;6u to the client program depending on kitty version. Presumably you have nvim configured to respond to eihter of those by quitting.

**Reply by omentic** [reply]
Interesting, my shell integration doesn't seem to be fully working: https://asciinema.org/a/HfkLeakoQj3iURN0t9K2bfXnX
Although I definitely did notice the glitch free resizing and the cwd in titlebar a few updates ago.

Here's the relevant part of the `--debug-input` log: http://ix.io/3MEO
Nothing looks particularly unusual - although I'm not quite sure what to look for.
I'm running kitty 0.24.1.

**Reply by kovidgoyal** [reply]
Doesnt look like you are running 0.24.1 or you have remapped ctrl+shift+z

From the log:
```
Press xkb_keycode: 0x34 clean_sym: z composed_sym: Z mods: ctrl+shift glfw_key: 122 (z) xkb_key: 122 (z) shifted_key: 90 (Z)
on_key_input: glfw key: 0x7a native_code: 0x7a action: PRESS mods: ctrl+shift text: '' state: 0 sent encoded key to child
```

the key is being sent to the child process, as I surmised.

**Reply by omentic** [reply]
I'm definitely running 0.24.1: `kitty 0.24.1 created by Kovid Goyal`. From the asciinema I posted (https://asciinema.org/a/HfkLeakoQj3iURN0t9K2bfXnX), it seems that the other integration commands may do something similar. 

Which is weird - my `.bashrc` has 
```bash
# BEGIN_KITTY_SHELL_INTEGRATION
if test -n "$KITTY_INSTALLATION_DIR" -a -e "$KITTY_INSTALLATION_DIR/shell-integration/bash/kitty.bash"; then source "$KITTY_INSTALLATION_DIR/shell-integration/bash/kitty.bash"; fi
# END_KITTY_SHELL_INTEGRATION`
```
appended to it, and I don't have `shell_integration` anywhere in kitty.conf. I'll go check through my Neovim settings for a `<C-S-Z>` remap, but since it didn't happen in a different terminal emulator, I suspect there'll be nothing there.

**Reply by omentic** [reply]
So it seems like `Ctrl+Z`/`Ctrl+Shift+Z` backgrounds Vim when in normal by default, which explains what the "crash" is. One mystery solved! But it shouldn't do that when in insert mode (when I usually accidentally press it) - and it doesn't in other terminal emulators, only kitty.

Seems that kitty sends `^Z` for `Ctrl+Z` and `^[[122;6u` for `Ctrl+Shift+Z`, while some others send `^Z` for both. `^[[122;6u` somehow breaks out of insert mode and then sends a `^Z` signal.

I guess this is now a different question - why my shell integration isn't fully integrated? It certainly looks like it's sending an escape code the escape code just doesn't do anything.

**Reply by trygveaa** [reply]
What you show in the first asciinema is what would happen if you press ctrl-z in nvim. It's put in the background, and to get it back you can run `fg` in the shell. I'm not sure why kitty sends ctrl-z on ctrl-shift-z, you should check your active config in kitty by pressing ctrl-shift-F6.

This is also exactly what's happening for me if I try it in alacritty, because alacritty sends ctrl-z when you press ctrl-shift-z, so not sure why it would not happen for you there.

**Comment by kovidgoyal** [comment]
On Tue, Jan 18, 2022 at 03:03:03AM -0800, j-james wrote:
> So it seems like `Ctrl+Z`/`Ctrl+Shift+Z` backgrounds Vim when in normal by default, which explains what the "crash" is. One mystery solved! But it shouldn't do that when in insert mode (when I usually accidentally press it) - and it doesn't in other terminal emulators, only kitty.
> 
> Seems that kitty sends `^Z` for `Ctrl+Z` and `^[[122;6u` for `Ctrl+Shift+Z`, while some others send `^Z` for both. `^[[122;6u` somehow breaks out of insert mode and then sends a `^Z` signal.
> 
> I guess this is now a different question - why my shell integration isn't fully integrated?

I told you already, you have remapped ctrl+shift+z in your kitty.conf

**Reply by omentic** [reply]
I don't have `ctrl+shift+z` mapped to anything in my kitty.conf, or any mappings involving the `z` key at all.

**Reply by page-down** [reply]
Run `kitty --config=NONE` and see if you still have the same problem?

**Reply by omentic** [reply]
I don't! That narrows it down a bunch, thanks! I still can't see any lines that might causing it though: http://ix.io/3MEZ

**Reply by page-down** [reply]
```text
kitty_mod ctrl+alt
```

This means that the default shortcuts for shell integration also uses the kitty_mod you configured.

I suggest that you keep only the options that you know and configured.

**Comment by ad-on-is** [comment]
I'm having the exactly same issue with kitty 0.37. I just switched from VSCode to neovim, and out of habit, I too often times hit ctrl+shift+z.

```
kitty --version    
kitty 0.37.0 created by Kovid Goyal
```
As can be seen in the screen capture, I ran kitty and neovim without any configs, and after hitting ctrl+shift+z neovim just quits.

https://github.com/user-attachments/assets/43e9a99f-a8a8-4cc0-934b-ee4cbb0be858

**Reply by kovidgoyal** [reply]
Does not replicate for me with

kitty --config=NONE nvim --clean
